File: Sort/QuickSort.java
import java.util.*;
class QuickSort{
    static int partition(int a[],int low,int high){
        int i=low-1;
        int pivot=a[high];

        for(int j=low;j<high;j++){
            if(a[j]<=pivot){
                i++;
                int temp=a[i];
                a[i]=a[j];
                a[j]=temp;
            }
        }

        int temp1=a[i+1];
        a[i+1]=a[high];
        a[high]=temp1;

        return i+1;
    }
    static void quickSort(int a[],int low,int high){
        if(low<high){
            int pi=partition(a, low, high);
            quickSort(a, low, pi-1);
            quickSort(a, pi+1, high);
        }
    }
    static void printArray(int a[]){
        for(int i=0;i<a.length;i++)
            System.out.print(a[i]+"\t");
    }
    public static void main(String[] args) {
        int a[] ={0,0,2,1};
        quickSort(a, 0, a.length-1);
        printArray(a);
    }
}

File: Sort/QuickSort/quicksort.cpp
#include <iostream>
#include<bits/stdc++.h>
#define fastio ios_base::sync_with_stdio(0);cin.tie(NULL);cout.tie(NULL)
using namespace std;
int partition(int a[],int low,int high){
	int pivot=a[high];
	int i=low-1;
	for(int j=low;j<=high-1;j++){
		if(a[j]<=pivot){
			i++;
			swap(a[i],a[j]);
		}
	}
	swap(a[i+1],a[high]);
	return i+1;
}
void quickSort(int a[],int low,int high){
	if(low<high){
		int pi=partition(a,low,high);
		quickSort(a,low,pi-1);
		quickSort(a,pi+1,high);
	}
}
void printArray(int *a,int n){
	for(int i=0;i<n;i++)
		std::cout<<a[i]<<" ";
}
int main(){
	fastio;
	int a[]={2,1,4,156,4,2,1,3,23,12,3};
	int n=sizeof(a)/sizeof(a[0]);
	int low=0;
	int high=n-1;
	quickSort(a,low,high);
	printArray(a,n);
	return 0;
}